main.cpp

```diff
@@ -423,6 +397,83 @@ IntType &IntType::divide(int rhs)
     return *this;
 }
 
+IntType &IntType::pow(int exp)
+{
+    return powInternal(exp);
+}
+
+IntType &IntType::pow(const FloatType &exp)
+{
+    return powInternal(static_cast<int>(exp));
+}
+
+IntType &IntType::pow(const DoubleType &exp)
+{
+    return powInternal(static_cast<int>(exp));
+}
+
+IntType &IntType::pow(const IntType &exp)
+{
+    return powInternal(static_cast<int>(exp));
+}
+
+IntType &IntType::powInternal(int exp)
+{
+    *value = static_cast<int>(std::pow(*value, exp));   
+    return *this;
+}
+
+// Point
+struct Point
+{
+    Point(float a, float b);
+    Point(const FloatType &a, const FloatType &b);
+    Point(const DoubleType &a, const DoubleType &b);
+    Point(const IntType &a, const IntType &b);
+
+    Point& multiply(float m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point &multiply(const FloatType &m);
+    Point &multiply(const DoubleType &m);
+    Point &multiply(const IntType &m);
```

Comment:
Don't put `&` next to the function name. 
put it next to the return type.
It looks like you're taking the address of the function, instead of returning a reference.
- [ ] revise